charset/charset.go

```diff
@@ -52,6 +53,15 @@ var charsetInfos = []*Charset{
 	{CharsetBin, CollationBin, make(map[string]*Collation), "binary", 1},
 }
 
+// All the names supported collations should be in the following table.
+var supportedCollationNames = map[string]interface{}{
```

Comment:
you can use `map[string]struct{}` instead, struct{} is zero costs.

Reply:
thank you!